tests/integration/features/features_int_test.go

```diff
@@ -127,6 +135,119 @@ var _ = Describe("preconditions", func() {
 
 })
 
+var _ = Describe("Manifest sources", func() {
+	Context("using various manifest sources", func() {
+
+		var (
+			objectCleaner *envtestutil.Cleaner
+			dsciSpec      *dscv1.DSCInitializationSpec
+			namespace     = "default"
+		)
+
+		BeforeEach(func() {
+			objectCleaner = envtestutil.CreateCleaner(envTestClient, envTest.Config, timeout, interval)
+			dsciSpec = newDSCInitializationSpec(namespace)
+		})
+
+		It("should be able to process an embedded template from the default location", func() {
+			// given
+			ns := createNamespace("service-ns")
+			Expect(envTestClient.Create(context.Background(), ns)).To(Succeed())
+			defer objectCleaner.DeleteAll(ns)
+
+			serviceMeshSpec := &dsciSpec.ServiceMesh
+			serviceMeshSpec.ControlPlane.Namespace = "service-ns"
+
+			createService, err := feature.CreateFeature("create-control-plane").
+				For(dsciSpec).
+				Manifests(path.Join(templatesDir, "serverless", "serving-istio-gateways", "local-gateway-svc.tmpl")).
+				UsingConfig(envTest.Config).
+				Load()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			By("verifying service is created after applying feature", func() {
+				Expect(createService.Apply()).To(Succeed())
+				service, err := getService("knative-local-gateway", "service-ns")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(service).ToNot(BeNil())
+			})
+		})
+
+		It("should be able to process an embedded yaml file from the default location", func() {
+			// given
+			ns := createNamespace("knative-serving")
+			Expect(envTestClient.Create(context.Background(), ns)).To(Succeed())
+			defer objectCleaner.DeleteAll(ns)
+
+			createGateway, err := feature.CreateFeature("create-gateway").
+				For(dsciSpec).
+				Manifests(path.Join(templatesDir, "serverless", "serving-istio-gateways", "istio-local-gateway.yaml")).
+				UsingConfig(envTest.Config).
+				Load()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			By("verifying gateway is created after applying feature", func() {
+				Expect(createGateway.Apply()).To(Succeed())
+				gateway, err := getGateway(envTest.Config, "knative-serving", "knative-local-gateway")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(gateway).ToNot(BeNil())
+			})
+		})
+
+		It("should be able to process an embedded file from a non default location", func() {
+			createNs, err := feature.CreateFeature("create-ns").
+				For(dsciSpec).
+				ManifestSource(testEmbeddedFiles).
+				Manifests(path.Join(templatesDir, "namespace.yaml")).
+				UsingConfig(envTest.Config).
+				Load()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			By("verifying namespace is created after applying feature", func() {
+				Expect(createNs.Apply()).To(Succeed())
+				namespace, err := getNamespace("embedded-test-ns")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(namespace).ToNot(BeNil())
+			})
```

Comment:
Same as before. We are missing `then` block.

Reply:
740741b